File: w2/db_query/backend/app/models/nl_query.py
from typing import Any, List, Optional
from pydantic import BaseModel, Field


class NLQueryRequest(BaseModel):
    connection_id: int = Field(..., alias="connectionId")
    prompt: str
    api_key: str | None = Field(None, alias="apiKey")

    class Config:
        populate_by_name = True


class NLQueryResponse(BaseModel):
    generated_sql: str = Field(..., alias="generatedSql")
    columns: List[str]
    rows: List[List[Any]]
    limit_added: bool = Field(False, alias="limitAdded")
    message: Optional[str] = None

    class Config:
        populate_by_name = True

File: w2/db_query/frontend/src/services/nl_query.ts
import { apiClient } from './api'

export interface NLQueryPayload {
  connectionId: number
  prompt: string
  apiKey?: string
}

export interface NLQueryResult {
  generatedSql: string
  columns: string[]
  rows: any[][]
  limitAdded: boolean
  message?: string
}

export async function nlQuery(payload: NLQueryPayload): Promise<NLQueryResult> {
  const { data } = await apiClient.post('/nl-query', payload)
  return data
}

